SwiftVoiceCallKitQuickstart/ViewController.swift

```diff
@@ -210,29 +210,32 @@ class ViewController: UIViewController, PKPushRegistryDelegate, TVONotificationD
         routeAudioToSpeaker()
     }
     
-    func callDidDisconnect(_ call: TVOCall) {
-        NSLog("callDidDisconnect:")
-
-        performEndCallAction(uuid: call.uuid)
-
-        self.call = nil
-        self.callKitCompletionCallback = nil
+    func call(_ call: TVOCall, didFailToConnectWithError error: Error) {
+        NSLog("Call failed to connect: \(error.localizedDescription)")
         
-        self.placeCallButton.setTitle("Place Outgoing Call", for: .normal)
+        performEndCallAction(uuid: call.uuid)
+        callDisconnected()
+    }
+    
+    func call(_ call: TVOCall, didDisconnectWithError error: Error?) {
+        if (error != nil) {
```

Comment:
We could be more Swift idiomatic here if we wanted. Something like (haven't tested, just speculating):

```
if let error = error {
    NSLog("Call failed: \(error.localizedDescription)")

    if let completion = self.callKitCompletionCallback {
        completion(false)
    }
} else {
    NSLog("Call disconnected"
}
```

Reply:
This makes sense to me, and your suggestion would definitely be more idiomatic.

Functionality wise - since this callback is always fired after `callDidConnect` you should not fire the completion block a 2nd time with a failure.